compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -349,7 +349,11 @@ object TypeOps:
       val doms = dominators(commonBaseClasses, Nil)
       def baseTp(cls: ClassSymbol): Type =
         tp.baseType(cls).mapReduceOr(identity)(mergeRefinedOrApplied)
-      doms.map(baseTp).reduceLeft(AndType.apply)
+      def meet(tp1: Type, tp2: Type) =
+        if !tp1.exists then tp2
```

Comment:
I don't understand why this is needed, we do:
```scala
val commonBaseClasses = tp.mapReduceOr(_.baseClasses)(intersect)
```
(btw, how is this different from `tp.baseClasses` ?)
So I would expect `tp.baseType` on any of the common base classes to exist, but in fact the base classes of `Tuple1[(Axis, Int)] | ((Axis, Int), (Axis, Int))` contains `Product2`, even though `Tuple1` does not extend `Product2`, so I think there's a bug in how we compute base classes for unions, either here or in OrType#baseClasses.

Reply:
> (btw, how is this different from tp.baseClasses ?)

Turns out it's different for `Null`. Since `Null` is also outside the inheritance hierarchy.